lib/ex_aws/s3/impl.ex

```diff
@@ -441,8 +441,28 @@ defmodule ExAws.S3.Impl do
     |> Parsers.parse_list_parts
   end
 
+  def presigned_url(client, http_method, bucket, object, opts \\ []) do
+    expires_in = Keyword.get(opts, :expires_in, 3600)
+    virtual_host = Keyword.get(opts, :virtual_host, false)
+    raise_if_exceeds_one_week(expires_in)
```

Comment:
Hey, so I would prefer if we did not raise here, and instead returned an {:error, reason} tuple. Correspondingly we should have a return of `{:ok, url}` on success.